### PR TITLE
Verify that email addresses are valid before saving request documents

### LINKED ISSUE #94: Check that email addresses look valid in `cyhy-simple` and `cyhy-import`
# 💡 Summary #

We should verify email addresses in [`cyhy-simple`](https://github.com/cisagov/cyhy-core/blob/6ddfa8852f741969078217984eeebeef6def83de/bin/cyhy-simple#L181-L195) and [`cyhy-import`](https://github.com/cisagov/cyhy-core/blob/6ddfa8852f741969078217984eeebeef6def83de/bin/cyhy-import#L100-L146), or possible just when [the request document is saved in the database](https://github.com/cisagov/cyhy-core/blob/6ddfa8852f741969078217984eeebeef6def83de/cyhy/db/database.py#L1286-L1294).

## Motivation and context ##

I got this error for an organization when sending out the CyHy reports this week:
```console
cyhy-mailer-mailer-1  | botocore.exceptions.ClientError: An error occurred (InvalidParameterValue) when calling the SendRawEmail operation: Domain contains illegal character
```
The issue turned out to be a duplicated at sign (`@`) in the distribution email address.  A simple regex would prevent errors like this from happening in the future.

### PR DESCRIPTION
**Pull request by jsf9k**
## 🗣 Description ##

This pull request modifies the code to verify that email addresses are valid before saving request documents.

## 💭 Motivation and context ##

Resolves #94.

## 🧪 Testing ##

I deployed this change to `database1.prod-a.cyhy`, exported a customer with `cyhy-export`, then tinkered with the email addresses and attempted to import said customer via `cyhy-import`.  WHen I used an valid email address the import was successful and when I used an invalid email address the import failed with an error similar to the following:
```console
$ cyhy-import --force REDACTED.json
Traceback (most recent call last):
  File "/usr/local/bin/cyhy-import", line 186, in <module>
    main()
  File "/usr/local/bin/cyhy-import", line 177, in main
    success = import_file(db, args["FILE"], args["--force"], args["--init-stage"])
  File "/usr/local/bin/cyhy-import", line 158, in import_file
    return import_request(db, request, filename, force, init_stage)
  File "/usr/local/bin/cyhy-import", line 145, in import_request
    doc.save()
  File "/usr/local/lib/python2.7/dist-packages/cyhy/db/database.py", line 1310, in save
    raise ValueError(email_address + " is not a valid email address")
ValueError: redacted@@redacted.com is not a valid email address
```

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All new and existing tests pass.

## ✅ Post-merge checklist ##

- [x] Deploy this change to `database1.prod-a.cyhy`.

### CODE REVIEW

**Review by dv4harr10 — APPROVED** [review]

**Review by dav3r — APPROVED** [review]
Excellent! 👍

**Comment by jsf9k** [comment]
I'll wait for @mcdonnnj's review before merging this one.

File: cyhy/db/database.py
Comment (mcdonnnj) [inline cyhy/db/database.py]: Since we're not doing anything special with the default return of `None` I think this would be cleaner.

```suggestion
        contacts = self["agency"].get("contacts", [])
        for contact in contacts:
            email_address = contact.get("email")
            if email_address is not None:
                # This should be modified to use fullmatch() when this code is
                # ported to Python 3.
                if not self.__EmailAddressRegex.match(email_address):
                    raise ValueError(email_address + " is not a valid email address")
```

**Review by mcdonnnj — APPROVED** [review]
LGTM with one suggestion for your consideration.

**Pull request merged**
